Sources/SwiftDocCTestUtilities/XCTestCase+TemporaryDirectory.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+
+// These helpers methods exist to put temp files for different test executions in different locations when running in Swift CI.
+
+public extension XCTestCase {
+    
+    @available(*, deprecated, message: "Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
```

Comment:
What's blocking us from moving away from `NSTemporaryDirectory` everywhere? I find it odd to mark an API as deprecated which still has usages within the repo, we're going to see deprecation warnings for those.

Reply:
This function only shadows calls made from within test cases. I've updated all of those, so this doesn't add any deprecation warnings.

Reply:
Can we do this for FileManager.default.temporaryDirectory as well? Not sure if Swift will allow that...

Reply:
No. I'm not aware of a way tp shadow a property on another type like that.

Reply:
I gave this a go- it looks like you can just redefine it but you need to define it in each test framework so that the compiler knows which overload to favor. (It uses an overload in the local module over one in the imported module.)

```
extension FileManager {
    @available(*, deprecated, message: "Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
    var temporaryDirectory: URL {
        XCTFail("Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
        return Foundation.NSTemporaryDirectory()
    }
}
```

Reply:
I actually just added this in this branch and found a couple uses:

```
/Users/demo/Developer/swift-docc/Tests/SwiftDocCUtilitiesTests/ConvertActionTests.swift:1843:58: 'temporaryDirectory' is deprecated: Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.
/Users/demo/Developer/swift-docc/Tests/SwiftDocCUtilitiesTests/ConvertActionTests.swift:1910:64: 'temporaryDirectory' is deprecated: Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.
/Users/demo/Developer/swift-docc/Tests/SwiftDocCTests/Converter/DocumentationConverterTests.swift:29:43: 'temporaryDirectory' is deprecated: Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.
```

Reply:
Great. I didn't think that would work. I'll incorporate the changes. Thanks!